### PR TITLE
If stock changes between page load and editing, reject stock changes

### LINKED ISSUE #15818: WooCommerce overrides stock instead of incrementing/decrementing it
## Prerequisites

- [ x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [ x] The issue still exists against the latest `master` branch of WooCommerce
- [ x] This is not a usage question (Those should be directed to the [community](https://wordpress.org/support/plugin/woocommerce), unless this is a question about a premium plugin in which you should [use the helpdesk](https://woocommerce.com/my-account/tickets/) for official extensions or contact the author of 3rd party extensions)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1.Shopping manager opens Product page (ex. Jeans) and sees that there are 5 remaining in stock
2.A random user buys 1 pair of Jeans, causing the stock to decrement to 4. But the admin still sees 5 on his Product page (still open and not refreshed)
3.The admin, in order to add 2 pairs of Jeans, overwrites the 5 and enters 7, and Saves.

## Expected behavior and actual behavior

When I follow those steps, I see that now the stock for the Jeans is 7.

I was expecting to see 6 because 1 of them was bought during the admin modification.
I was expecting Woocommerce NOT to override the 5, but to do 7 minus 5 equals 2, and increment the stock by 2.

I would solve it this way, OR the product page could display the actual stock via ajax in real-time, NOT allowing it to be modified, but instead using an "increment by/decrement by" button.

## Environment

<details>
Latest version of WooCommerce, latest version of WordPress. 
</details>

## Isolating the problem

- [x ] This bug happens with only WooCommerce plugin active
- [ x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [ x] I can reproduce this bug consistently

### PR DESCRIPTION
**Pull request by mikejolley**
Fixes #15818

To test, load 2 tabs of the same edit product page. Edit one, then try to edit the 2nd one. The changes will be rejected with a notice.

### CODE REVIEW

File: includes/admin/meta-boxes/views/html-variation-admin.php [hunk 1]
```diff
@@ -170,6 +170,8 @@
 							'wrapper_class'     => 'form-row form-row-first',
 						) );
 
+						echo '<input type="hidden" name="variable_original_stock[' . $loop . ']" value="' . esc_attr( wc_stock_amount( $variation_object->get_stock_quantity( 'edit'  ) ) ) . '" />';
```
Comment (claudiulodro) [inline includes/admin/meta-boxes/views/html-variation-admin.php]: Too many spaces after `'edit'`.

File: includes/admin/meta-boxes/class-wc-meta-box-product-data.php [hunk 2]
```diff
@@ -310,6 +310,17 @@ public static function save( $post_id, $post ) {
 		$classname    = WC_Product_Factory::get_product_classname( $post_id, $product_type ? $product_type : 'simple' );
 		$product      = new $classname( $post_id );
 		$attributes   = self::prepare_attributes();
+		$stock        = null;
+
+		// Handle stock changes.
+		if ( isset( $_POST['_stock'] ) ) {
+			if ( isset( $_POST['_original_stock'] ) && wc_stock_amount( $product->get_stock_quantity( 'edit' ) ) !== wc_stock_amount( $_POST['_original_stock'] ) ) {
+				WC_Admin_Meta_Boxes::add_error( sprintf( __( 'The stock has not been updated because the value has changed since editing. Product %d has %d units in stock.', 'woocommerce' ), $product->get_id(), $product->get_stock_quantity( 'edit' ) ) );
```
Comment (claudiulodro) [inline includes/admin/meta-boxes/class-wc-meta-box-product-data.php]:
```
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 318 | ERROR | [x] Multiple placeholders should be ordered. Expected
     |       |     '%1$d, %2$d', but got %d,%d.
     |       |     (WordPress.WP.I18n.UnorderedPlaceholders)
 410 | ERROR | [x] Multiple placeholders should be ordered. Expected
     |       |     '%1$d, %2$d', but got %d,%d.
     |       |     (WordPress.WP.I18n.UnorderedPlaceholders)
```

**Comment by claudiulodro** [comment]
Just a few formatting changes needed. I like this solution, though.

**Comment by UserGalileo** [comment]
Personally I don't mind this solution, as it solves the logical problem and avoids the error. However, I feel that for a Shopping manager it wouldn't be so good in terms of user experience: when the system throws the error, I (shopping manager) think "Hey, it's good that the system alerts me, but why doesn't it solve the problem for me?" :) I mean, it could just make a subtraction and update the stock value for me, I don't think it'd be over-complicated. And, with your proposal, if I make some modifications to the product itself (description, title or any other field) I'd have to discard all the changes I've made in order to re-enter the page and see the stock updated!
What do you think about it?

File: includes/admin/meta-boxes/class-wc-meta-box-product-data.php (hunk 2 above)
Comment (claudiosanches) [inline includes/admin/meta-boxes/class-wc-meta-box-product-data.php]: Requires comments for translators too.

```php
/* translators: 1: product ID 2: quantity in stock */
```

**Pull request merged**